Lib/zipfile/__init__.py

```diff
@@ -487,7 +487,7 @@ def FileHeader(self, zip64=None):
 
     def _encodeFilenameFlags(self):
         try:
-            return self.filename.encode('ascii'), self.flag_bits
+            return self.filename.encode('cp437'), self.flag_bits
```

Comment:
This would need to add the removal of the UTF-8 bit from flag_bits again... which would re-create this regression.  https://github.com/python/cpython/issues/95463

I'll follow up on the issue.